### PR TITLE
 fixing issue #126 stitchingport on one or more than one 2-ends broad…

### PR DESCRIPTION
**Pull request by YufengXin**
…cast links, and manifest drawing correctly

### CODE REVIEW

**Comment by hinchliff** [comment]
@YufengXin: could you add a test to cover this change?  The simplest test to add would probably be to add another test to [OrcaRegressionTest.java](https://github.com/RENCI-NRIG/orca5/blob/orca-5.1.4/controllers/xmlrpc/src/test/java/orca/controllers/xmlrpc/OrcaRegressionTest.java), but you don't have to do it that way.

**Comment by YufengXin** [comment]
Sure. Which subdirectory should I use, TS2?

**Comment by hinchliff** [comment]
TS2 would be fine, since this is a test very similar to TS2-9.

File: embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java [hunk 1]
```diff
@@ -957,6 +957,7 @@ public void createManifest(NetworkElement requestElement,OntModel manifestModel,
                 boolean isStitchingDomain = NdlCommons.isStitchingNodeInManifest(start.getResource());
                 link_url = start.getURI();
                 if(isStitchingDomain){
+                	link_url = start.getName();
```
Comment (hinchliff) [inline embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java]: It looks like the `source domain` and `destination domain` code here and below should be refactored to reduce code duplication.

**Comment by YufengXin** [comment]
Agreed, go ahead

In a few cases, in early days, I separated the source and destination on purpose for easier debugging in short period of time-:)

-Yufeng

**Comment by hinchliff** [comment]
@YufengXin I'm not sure if you saw my message on Slack, but the test TS2-14 is currently failing. You might be using a domain that I don't have turned on in the test environment?

If you could restrict testing to only the domains populated in [MockOrcaServiceManager.populateResourceMap()](https://github.com/RENCI-NRIG/orca5/blob/orca-5.1.4/controllers/xmlrpc/src/test/java/orca/controllers/mock/MockOrcaServiceManager.java#L42) that would be best for the overall test performance

Both of the tests that you have added so far only exercise the [`source domain`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L955) path in InterDomainHandler.  Would it be possible to add a StitchPort test that exercises the [`destination domain`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L1046) path?

There are some minor differences between those two code paths that I will need to understand better before we can refactor this mostly duplicated code.

* The code you added at [InterDomainHandler.java#L1052](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L1052) should probably use `next_Hop.getName()` ?
* Should the `inDomainProperty` of `link_ont` be set to [`start.getResource()`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L991) or [`link_ont`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L1083) or `next_Hop.getResource()`? Should it start or next based on source vs. destination? Or should it always be link_ont?
* Should the `destination domain` path set the [`inRequestNetworkConnection`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L1000) property?
* Should the property `NdlCommons.RDF_TYPE` of `NdlCommons.interfaceOntClass` that is set in the[`destination domain`](https://github.com/RENCI-NRIG/orca5/blob/issue126-bc-stitchingport/embed/src/main/java/orca/embed/cloudembed/controller/InterDomainHandler.java#L1091) path also be set in the `source domain` path?

**Comment by hinchliff** [comment]
@YufengXin let me know if I made valid assumptions in c9cd2c9d9e319b7cd96de37f78191bdb9539acb6.  A test that covers the `destination domain` path would be useful.

**Comment by YufengXin** [comment]
Thanks, I’ll take a closer look. Normally the source and destination are random for a path. But there are two corner cases for the manifest generation of the edge being: (1) stitching port; (2) inter-rack MP when one edge is that Departure dr. root. I forgot if I made them explicitly to be always the source (or destination).

-Yufeng

**Pull request merged**